tests/commonUtils.sh

```diff
@@ -44,19 +46,62 @@ function compare_output() {
     fi
 }
 
-function modify_rules_namespace(){
-    print_block_echo "Modifying rules to match namespace"
-    find $SCRIPTDIR/apps/bookinfo/rules/ -type f -print0 \
-      | xargs -0 sed -i "s/_CHANGEME_/$NAMESPACE/g" \
-      || error_exit 'Could not modify namespace rules'
+function apply_patch() {
+    local src=${1}
+    local dif=${2}
+    local dest=${3}
+
+    patch ${src} -i ${dif} -o ${dest} -R \
+      || error_exit "Could not apply patch ${dif} on ${src}"
+}
+
+function apply_patch_in_dir() {
+    local diff_dir=${1}
+    local src_dir=${2}
+    local dest_dir=${3}
+    local ext=${4}
+    local files=($(find "${diff_dir}" -maxdepth 1 -type f -name '*.diff'))
+
+    for dif in ${files[@]}; do
+      local filename=${dif##*/}
```

Comment:
What does the above line do ? Bash has a bunch of vodoo stuff that not everyone knows:). So it might be good practice if we add comments to such constructs so that someone else could debug when needed.